src/style_context.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright 2013-2020, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use glib::object::IsA;
+use glib::translate::*;
+use gtk_sys;
+use libc::c_void;
+use pango::FontDescription;
+use pango_sys::PangoFontDescription;
+use StyleContext;
+
+pub trait StyleContextExtManual: 'static {
+    fn get_font(&self) -> Option<FontDescription>;
```

Comment:
Why don't we bind `gtk_style_context_get_property()` btw? That would be the generic getter that allows to get any kind of property by string name

Reply:
It was ignored with `get_style_property` in 2016 (4e6d5db1991bdd923069d802d3250ea0ea15f655), now it works and we can try to de-ignore it and others

Reply:
You'll add it as part of this PR?

Reply:
I tried and found that most of ignored functions generated like this.
@sdroege What you think, we can leave as is or implement it manually as -> glib::Value ?
```
    fn style_get_property(&self, property_name: &str, value: &mut glib::Value) {
        unsafe {
            gtk_sys::gtk_widget_style_get_property(
                self.as_ref().to_glib_none().0,
                property_name.to_glib_none().0,
                value.to_glib_none_mut().0,
            );
        }
    }
```

Reply:
Added temporary commit with generation results

Reply:
As `-> glib::Value` or `-> Option<glib::Value>` (or `Result`) :(

Going with an out parameter is problematic here safety-wise, and also a bit annoying usage-wise.

Reply:
Ok I will implement it manually.

Reply:
For fix gir to generate it right

Reply:
Seems gir-file also somewhat wrong:
gtk_style_context_get_property generate return,
```
          <parameter name="value" direction="out" caller-allocates="1" transfer-ownership="full">
            <doc xml:space="preserve" filename="gtk/gtkstylecontext.c" line="797">return location for the style property value</doc>
            <type name="GObject.Value" c:type="GValue*"/>
          </parameter>
```
gtk_style_context_get_style_property: generate &mut Value
```
          <parameter name="value" transfer-ownership="none">
            <doc xml:space="preserve" filename="gtk/gtkstylecontext.c" line="1727">Return location for the property value</doc>
            <type name="GObject.Value" c:type="GValue*"/>
          </parameter>
```

Reply:
Indeed. Can you create issues for that against gtk, or maybe even a MR fixing the annotations? :)

Reply:
And maybe fix the .gir file for now so that we can autogenerate these already

Reply:
@sdroege Seems no such problem in gtk4, most functions was just removed (Ex. 
https://github.com/GNOME/gtk/commit/632fab0ca702b600fde492319ab0aa2d9d3bbd08)

You don't know, is https://gitlab.gnome.org/GNOME/gtk/tree/gtk-3-24 good for adding PR against it or there other branch for last gtk3

Reply:
Upstream issue https://gitlab.gnome.org/GNOME/gtk/issues/2418 with link to unchecked fix

Reply:
Try your look with a MR against the GTK 3 branch. Maybe there will be another bugfix release with that :)

Reply:
I don't know what branch is "dev" for gkt3, so just selected last version